Translations update from Weblate

tinect opened the pull request:
Translations update from [Weblate](http://weblatehome.tinect.de/projects/shopware-platform/storefront/) for shopware-platform/storefront.

bcremer commented:
There is no need for such an event because the `StoreFrontCriteriaFactory` implements an interface (`Shopware\Bundle\SearchBundle\StoreFrontCriteriaFactoryInterface`) and is injected via the DI Container (id:  `shopware_search.store_front_criteria_factory`).

That given the `StoreFrontCriteriaFactory` can be decorated.

```
class Shopware_Plugins_Frontend_SwagExample_Bootstrap extends Shopware_Components_Plugin_Bootstrap
{
    public function install()
    {
        $this->subscribeEvent(
            'Enlight_Bootstrap_AfterInitResource_shopware_search.store_front_criteria_factory',
            'decorateService'
        );

        return true;
    }

    public function decorateService()
    {
        $coreService = $this->container->get('shopware_search.store_front_criteria_factory');

        $decorator = new StoreFrontCriteriaFactoryDecorator($coreService);

        $this->container->set('shopware_search.store_front_criteria_factory', $decorator);
    }
}
```

The Decorator Class:

```
<?php
namespace ShopwarePlugins\SwagExample;

use Enlight_Controller_Request_Request as Request;
use Shopware\Bundle\SearchBundle\StoreFrontCriteriaFactoryInterface;
use Shopware\Bundle\StoreFrontBundle\Struct\ShopContextInterface;

class StoreFrontCriteriaFactoryDecorator implements StoreFrontCriteriaFactoryInterface
{
    /**
     * @var StoreFrontCriteriaFactoryInterface
     */
    private $inner;

    /**
     * StoreFrontCriteriaFactory constructor.
     * @param StoreFrontCriteriaFactoryInterface $inner
     */
    public function __construct(StoreFrontCriteriaFactoryInterface $inner)
    {
        $this->inner = $inner;
    }

    /**
     * @inheritdoc
     */
    public function createBaseCriteria($categoryIds, ShopContextInterface $context)
    {
        $criteria = $this->inner->createBaseCriteria($categoryIds, $context);

        $criteria->addCondition(...);

        return $criteria;
    }

    /**
     * @inheritdoc
     */
    public function createSearchCriteria(Request $request, ShopContextInterface $context)
    {
        return $this->inner->createSearchCriteria($request, $context);
    }

    /**
     * @inheritdoc
     */
    public function createListingCriteria(Request $request, ShopContextInterface $context)
    {
        return $this->inner->createListingCriteria($request, $context);
    }

    /**
     * @inheritdoc
     */
    public function createAjaxSearchCriteria(Request $request, ShopContextInterface $context)
    {
        return $this->inner->createAjaxSearchCriteria($request, $context);
    }

    /**
     * @inheritdoc
     */
    public function createAjaxListingCriteria(Request $request, ShopContextInterface $context)
    {
        return $this->inner->createAjaxListingCriteria($request, $context);
    }

    /**
     * @inheritdoc
     */
    public function createAjaxCountCriteria(Request $request, ShopContextInterface $context)
    {
        return $this->inner->createAjaxCountCriteria($request, $context);
    }

    /**
     * @inheritdoc
     */
    public function createProductNavigationCriteria(Request $request, ShopContextInterface $context, $categoryId)
    {
        return $this->inner->createProductNavigationCriteria($request, $context, $categoryId);
    }
}
```

bcremer commented on engine/Shopware/Bundle/SearchBundle/StoreFrontCriteriaFactory.php:
```diff
@@ -96,6 +96,10 @@ public function createBaseCriteria($categoryIds, ShopContextInterface $context)
         );
+
+        $this->eventManager->notify('Shopware_SearchBundle_Create_Base_Criteria', [
+            'criteria' => $criteria
```
Please add `$categoryIds` as event argument.

roman-1983 commented:
I have added `$categoryIds` :)

bcremer commented:
Merged with commit https://github.com/shopware/shopware/commit/377bc159681b53ebc64ba429b831f45bde76aca1. Thank you @roman-1983 :+1:

tinect committed:
Translated using Weblate (German)

Currently translated at 100.0% (454 of 454 strings)

Translation: shopware-platform/storefront
Translate-URL: http://weblatehome.tinect.de/projects/shopware-platform/storefront/de/
